jdaviz/configs/mosviz/helper.py

```diff
@@ -180,6 +187,23 @@ def _update_spec1d_x_axis(self, change=None):
 
         self._update_in_progress = False
 
+    def _redshift_listener(self, msg):
+        '''Save new redshifts (including from the helper itself)'''
+        if msg.param == "redshift":
+            row = self.app.get_viewer('table-viewer').widget_table.highlighted
+            # NOTE: this updates the value in the table for the current row.  This
+            # in turn will feedback to call _apply_redshift_from_table and set
+            # the internal value.  When implementing the slider itself, this logic
+            # may need to be changed slightly to avoid a race condition.
+            self.update_column('Redshift', msg.value, row=row)
+
+    def _apply_redshift_from_table(self):
+        # NOTE: this only applies to mosviz.specviz.get_spectra(...)
+        # not mosviz.app.get_data_from_viewer(...).  This should be
+        # cleaned up when the slider is implemented in mosviz.
+        row = self.app.get_viewer('table-viewer').widget_table.highlighted
+        self.specviz._redshift = self.get_column('Redshift')[row]
```

Comment:
If you have line lists loaded and plotted, those should shift.  Otherwise, it will just set the value in the Spectrum1D when retrieved by `mosviz.specviz.get_spectra()` (although app-level helper functions may be soon to come when also hooking into the Spectrum2D)